### PR TITLE
Make uploadMultipleFilesToDisk() work with model events

### PR DESCRIPTION
**Pull request by pxpm**
### BEFORE - What was wrong? What was happening before this PR?

`uploadMultipleFilesToDisk` would not properly work with model events as it works with a mutator. 

In the mutator the result of `$this->attribute` when in a new entry would be an empty array no previous value exists in model (in creating for example), but in the `saving` model event, `$this->attribute` will result in `[{},{}]`, so it would endup trying to save `[{},{}, 'file1.pdf', 'file2.pdf']`

### AFTER - What is happening after this PR?

After this PR `uploadMultipleFilesToDisk` works with mutators and model events just fine.

### How did you achieve that, in technical terms?

Specifically check if there is already an key present in the model, if not it means we are creating a new model, so we are sure there is no previous files on disk.

### Is it a breaking change or non-breaking change?

I don't think it's breaking, as it would be the default behaviour for the mutator.

### How can we test the before & after?

```php
CRUD::field('text')->type('upload_multiple')->upload(true)->disk('public')->events([
            'saving' => function($entry) {
                $attribute_name = "text";
                $disk = "public";
                if(request('text')) {
                    $entry->uploadMultipleFilesToDisk(request('text'), $attribute_name, $disk, '');
                }
            }
        ]);
```

Will error before this PR and will work fine after.

### CODE REVIEW

**Comment by tabacitu** [comment]
@pxpm sorry, read this 3 times and still can't understand the issue at hand. I get that it has to do with `uploadMultipleFilesToDisk()` and model events, but... I can't follow the logic here. 

**Can you please rephrase the issue?** Remember [The Whole Fruit Manifesto](https://github.com/the-whole-fruit/manifesto), specifically:
- shorter sentences;
- set context;
- use bullet points or numbered lists if necessary;
- the subject of a sentence should be PERSON if possible; in our case, a developer or an admin;

Thanks!

**Comment by tabacitu** [comment]
Bump @pxpm - please see my comment above.

**Comment by pxpm** [comment]
Hey @tabacitu 

TLDR; model events and model mutators have different places in the entry lifecycle and yield different results for the same function depending.

Regarding this, I will try to rephrase so you can understand. 

At the moment, if the developer wants to use an `upload_multiple` field he should define the field in controller, cast the field to array in model, and create a model mutator. 

Sometime ago we introduced the `field events` aka `model events` defined on fields. One of the reasons for that was to have `kind of mutators` for subfields. 

So at this moment developer should be able to use the `events => saving` on the `upload_multiple` field (theoretically), to handle the file saving process without creating the mutator on model.

I said theoretically because our `uploadMultipleFilesToDisk` function only account for beeing called on the attribute mutator.
The first bit of the function is a check for `previous model value`, so calling `$model->upload_multiple_attribute` would return either: `null` when there was no previous value, otherwise the attribute value, string or already array.

If we do the same call in the `saving` event of the model, the `$model->upload_multiple_attribute` would not yield the expected null when empty, because at this time the attribute is already set on model from what was provided in the request. 

So by example, I am in `CreateOperation`, and I've submitted two new files.

- calling the function from the model mutator, the result of `$model->upload_multiple_attribute` value would be `null`.
- calling from the `saving` event:
it would be either `[{},{}]` if casting to array
image
 or the array of `UploadedFile` when not casting.
image

I've refactored the fix I proposed, since the first one was not reliable. Basically I resorted to always get the value of the previous db files from the `orginal` version of the model, disregarding if there were any changes in the meanwhile because what I want is the previous value of the model stored in database. 

This ensure that the upload field will work either in mutator or in the model events when using the `uploadMultipleFilesToDisk` function.
